core/state_transition.go

```diff
@@ -473,7 +475,7 @@ func (st *StateTransition) innerTransitionDb() (*ExecutionResult, error) {
 		// After EIP-3529: refunds are capped to gasUsed / 5
 		st.refundGas(params.RefundQuotientEIP3529)
 	}
-	if st.msg.IsDepositTx && rules.IsOptimismRegolith {
+	if st.msg.IsDepositTx && rules.IsOptimismRegolith || st.msg.IsMintTokenTx && rules.IsKromaBurgundy {
```

Comment:
For better readability,
```suggestion
	if (st.msg.IsDepositTx && rules.IsOptimismRegolith) || (st.msg.IsMintTokenTx && rules.IsKromaBurgundy) {
```